### PR TITLE
Consolidate all the jobs into a single pull_request_target based job

### PR DESCRIPTION
**Pull request by baijum**
Consolidate all the jobs into a single `pull_request_target` based job.
- Sanity Check
- Build and Verify
- Merge PR
- Release Charts (GitHub Release) and update `index.yaml`

### CODE REVIEW

File: .github/workflows/build.yml
Comment (pedjak) [inline .github/workflows/build.yml:109]: we need to checkout this branch in a separate folder, to avoid overriding the target checkout and potential attacks. Perhaps we can do it manually, using `git`.

File: .github/workflows/build.yml
Comment (pedjak) [inline .github/workflows/build.yml:111]: this not needed anymore I think, because there is just single workflow now.

File: .github/workflows/build.yml
Comment (pedjak) [inline .github/workflows/build.yml:124]: this step can be moved under sanity check job, right?

File: .github/workflows/build.yml
Comment (pedjak) [inline .github/workflows/build.yml:143]: IMHO, these two steps are not needed anymore.

File: .github/workflows/build.yml
Comment (pedjak) [inline .github/workflows/build.yml:156]: is this line really needed?

File: .github/workflows/build.yml
Comment (pedjak) [inline .github/workflows/build.yml:172]: is this line really needed?

**Review by pedjak — CHANGES_REQUESTED** [review]
let's collapse all jobs into single one, so that we avoid repeating steps (python install, for example) and make communication between steps easier.

File: .github/workflows/build.yml
Comment (baijum) [reply .github/workflows/build.yml:27]: I think we can keep sanity very minimal. There are more checks about `OWNERS` file and its content later in the `Build and Verify` section.
1. https://github.com/openshift-helm-charts/charts/tree/main/docs#pull-request-author-is-not-part-of-owners-file
2. https://github.com/openshift-helm-charts/charts/pull/112 (waiting for approval)

File: .github/workflows/build.yml
Comment (baijum) [reply .github/workflows/build.yml]: Removed. 76f652c5dcc1eb7d3443167b1c25c64c3fade50c

File: .github/workflows/build.yml
Comment (baijum) [reply .github/workflows/build.yml]: Removed. 76f652c5dcc1eb7d3443167b1c25c64c3fade50c

File: .github/workflows/build.yml
Comment (baijum) [inline .github/workflows/build.yml:234]: Probably this docker pull is not required here.

File: .github/workflows/build.yml
Comment (baijum) [reply .github/workflows/build.yml:106]: Now everything under a common job.

File: .github/workflows/build.yml
Comment (mmulholla) [inline .github/workflows/build.yml:201]: Do we still need this step?

Comment (mmulholla) [inline .github/workflows/build.yml:201]: Do we need this step?

Comment (baijum) [reply .github/workflows/build.yml:201]: Git commit will not work without this. Also, we need a way to identify who made the commit when looking at the git history.

**Comment by mmulholla** [comment]
@baijum @pedjak 
I am good to merge this and continue to make adjustements as needed.

**Comment by baijum** [comment]
> @baijum @pedjak
> I am good to merge this and continue to make adjustments as needed. It makes significant progress and auto-merge works.

I am fine with that approach. We can continue refactoring after this PR is merged. This will be helpful in my chart-testing PR.

**Comment by mmulholla** [comment]
Fixed a couple of insert errors in messages and collapsed to one commit. Looking good.

**Pull request merged**
